### PR TITLE
Add snapping info to the catch selected objects inspector

### PR DESCRIPTION
**Pull request by Darius-Wattimena**
In stable, catch mappers used the "x value" in the top right to determine how strong a walk, dash or hyperdash is to map somewhat consistently. This PR adds those values to the inspector.

In addition to being essential to mappers, it is also used quite a bit in modding. See for example [this mod post](https://osu.ppy.sh/beatmapsets/2237675/discussion/4754469/timeline#/4599778) where I suggested to use a "2.2x distance".

### CODE REVIEW

File: osu.Game.Rulesets.Catch/Edit/CatchHitObjectInspector.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                {
+                    double previousSnap = snapProvider.ReadCurrentDistanceSnap(precedingObject, firstSelectedHitObject);
+                    AddValue($"Previous: {previousSnap:#,0.##}x");
+                }
+
+                if (nextObject != null)
+                {
+                    double nextSnap = snapProvider.ReadCurrentDistanceSnap(lastSelectedHitObject, nextObject);
+                    AddValue($"Next: {nextSnap:#,0.##}x");
+                }
```
Comment (bdach) [inline osu.Game.Rulesets.Catch/Edit/CatchHitObjectInspector.cs:39]: I'm weirded out by this formatting. osu! does this:



Can we use the same but change the units to the `#.##x` relative thing? I'd rather be consistent here.

Comment (Darius-Wattimena) [reply osu.Game.Rulesets.Catch/Edit/CatchHitObjectInspector.cs:39]: I intially made it consistent to how the position is shown, so like this:

But consistency is probably better here 👍

**Comment by bdach** [comment]
I've pushed an extra change that turns off these displays if the previous/next object in question is a banana shower. Mainly because it was showing nonsensical stuff, but also because showing that kinda doesn't make sense to me (banana placement is RNG, and it's all bonus, so I'm not sure why anyone would care).

Other than that, looks good, although I will ask @peppy to confirm one thing here - are you OK with catch using `#.##x` object distances, rather than "playfield pixels" like osu! currently does? To me it makes more sense to use this sort of formatting in catch, given that "1.0x" in catch is the maximum walk speed, so having distances be relative to that walking range actually makes a lot of sense to me, while it didn't necessarily make much sense to me in osu!.

**Review by bdach — APPROVED** [review]

**Comment by peppy** [comment]
I think that makes sense, yes. We'll see how mappers fare with it.

**Review by peppy — APPROVED** [review]

**Pull request merged**